test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -514,6 +515,36 @@ var _ = ginkgo.Describe("end to end testing", func() {
 			gomega.Expect(*cs.Results[0].Rows[0].Columns[0].Number).To(gomega.Equal("2"))
 		})
 
+		ginkgo.It("list of WECs with available nginx deployment", func(ctx context.Context) {
+
+			availableNginxCEL := ksapi.Expression("obj.spec.replicas == returned.status.availableReplicas && obj.metadata.labels['app.kubernetes.io/name'] == 'nginx'")
```

Comment:
Why is the `&& obj.metadata.labels['app.kubernetes.io/name'] == 'nginx'` in here? The `obj.metadata.labels` is from the WDS and does not vary from WEC to WEC.

Reply:
The intention was to filter out all the objects in the WDS without this label(the test name says: `available nginx`) , and to exercise `obj.metadata` along with `obj.spec`

Reply:
The filtering wrt workload object is done prior. Remember that a CombinedStatus is already specific to a pair of workload object and StatusCollector.

If you want to test access to `obj.metadata` then I recommend doing it in a way that makes a difference to the resulting CombinedStatus.

Reply:
Put another way, it is the job of the BindingPolicy and the Binding to associate StatusCollectors with workload objects. The filter in a StatusCollectorSpec looking at data that is specific to one workload object but (potentially) varying from one WEC to another.

Reply:
ok. I removed this. Can't see a way to test an access to `obj.metadata`  in a way that makes a difference to the resulting CombinedStatus.

Reply:
We can pursue that additional testing in follow-on PRs.